### PR TITLE
`big-integer` shim testing on `android-jsc`

### PR DESCRIPTION
**Pull request by leotm**
**Description**

Testing `big-integer` shim on problematic `android-jsc` cc @Gudahtt
- https://github.com/MetaMask/metamask-mobile/pull/6221/commits/17d37881cf6d93d55d41c27d828042a1db3890ee
- https://github.com/MetaMask/metamask-mobile/pull/6221/commits/521417a4f64cd2878188976c237e33ebd856807a
- https://github.com/MetaMask/metamask-mobile/pull/6221/commits/d59853721e55d8acf36c9a6102c9d9f9f1470eca
- https://github.com/MetaMask/metamask-mobile/pull/6221/commits/64fb902082ae70f4d1fe13b35b787c981488d97b

since iOS JSC already natively fully supports big integers (`BigInt` and big integer literals like `1n`)
thus compat with `@ethereumjs/util`

whereas vanilla `android-jsc` (even on latest RN 0.71.6) throws
- e.g. `BigInt(1)`: ReferenceError: Can't find variable: BigInt
  - fixed by `big-integer` shim
- e.g. `1n`: SyntaxError: No identifiers allowed directly after numeric literal
  - not fixed by `big-integer` shim

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1959,6 +2005,15 @@
     async "^3.2.4"
     ethereum-cryptography "^1.1.2"
 
+"@ethereumjs/util@^8.0.5":
+  version "8.0.5"
+  resolved "https://registry.yarnpkg.com/@ethereumjs/util/-/util-8.0.5.tgz#b9088fc687cc13f0c1243d6133d145dfcf3fe446"
+  integrity sha512-259rXKK3b3D8HRVdRmlOEi6QFvwxdt304hhrEAmpZhsj7ufXEOTIc9JRZPMnXatKjECokdLNBcDOFBeBSzAIaw==
+  dependencies:
+    "@chainsafe/ssz" "0.9.4"
```
[the remark this answers is not in the document]
Comment (leotm) [reply yarn.lock:2013]: the shim's unsupported big int literals (e.g. `1n`) remain
but yeah not rly a prob long as our libs don't complain

[the remark this answers is not in the document]
Comment (leotm) [reply yarn.lock:2013]: testing `@ethereumjs/util` 8.0.6 with `arrToBufArr` https://github.com/MetaMask/metamask-mobile/pull/6221/commits/64fb902082ae70f4d1fe13b35b787c981488d97b throws

Screenshot 2023-04-27 at 14 14 38

which is true `node_modules/zlib` doesn't exist 🤔

and noticed
@ethereumjs/util/src/index.ts also exports * from './provider'
'./provider' imports fetch from 'micro-ftch'
'./provider' exports fetchFromProvider

[the remark this answers is not in the document]
Comment (leotm) [reply yarn.lock:2013]: just checked a fresh (RN 0.71.6) as well
switched off Hermes (for android-jsc)
added and ran `rn-nodeify --install http,https,zlib,events,util --hack --yarn` (minimal shims)
added and manually shimmed `big-integer`
added `@ethereumjs/util` 8.0.6 and tested `arrToBufArr`
which gets us past node missing shim issues like `zlib`
onto a new issue (potentially more):

rn-nodeify --install http,https,zlib,events,util --hack --yarn

**Comment by Gudahtt** [comment]
This should no longer be relevant now that we've upgraded React Native

**Pull request closed without merging**